### PR TITLE
fix(tooltip): Fix Tooltip component position and padding. also adds r…

### PR DESCRIPTION
**Pull request by pedrohmorais**
## Description
Fixes `Tooltip` component position and padding.
Adds regression test for this component.

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] IE 11
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
- [ ] Ux review validation

### CODE REVIEW

**Comment by rapahaeru** [comment]
In the tooltip examples tab, shows it on console.
image

File: stories/Tooltip/Tooltip.regression-test.story.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <Row style={{ padding: '50px 180px' }}>
+    <Col>
+      <Tooltip placement={direction} text="This is a hint." visible>
+        <Button>Hover me</Button>
+      </Tooltip>
+    </Col>
+  </Row>
+);
+
+directions.forEach(direction => {
```
Comment (rapahaeru) [inline stories/Tooltip/Tooltip.regression-test.story.jsx]: good

File: stories/Tooltip/Tooltip.story.jsx [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+import { Row, Col } from '../../components/Grid';
 import Tooltip from '../../components/Tooltip';
+import examples from './examples/examples';
+
+const TabExample = (
+  <Tab title="Examples">
+    <StoryContainer>
+      <Title as="h2">Examples</Title>
+      {examples.map(example => (
+        <Row key={example.code}>
```
Comment (rapahaeru) [inline stories/Tooltip/Tooltip.story.jsx]: Some how, its not working.
image

File: components/Tooltip/options.js [hunk 3]
```diff
@@ -46,22 +47,22 @@ const placementConfig = {
   tipPosition: {
     top: `
       left: ${PERCENTAGE_X}%;
-      bottom: ${PERCENTAGE_Y}%;
+      bottom: calc(${PERCENTAGE_Y}% + ${ARROW_SIZE + BOX_MARGIN}px);
```
[the remark this answers is not in the document]
Comment (pedrohmorais) [reply components/Tooltip/options.js]: It was overwritten by a new variable called `tipPositionDistance`

File: components/Tooltip/options.js [hunk 4]
```diff
@@ -46,22 +47,22 @@ const placementConfig = {
...
     `,
     right: `
-      left: ${PERCENTAGE_Y}%;
+      left: calc(${PERCENTAGE_Y}% + ${ARROW_SIZE + BOX_MARGIN}px);
       top: ${PERCENTAGE_X}%;
       transform: translateY(-${PERCENTAGE_X}%);
   `,
     left: `
-      right: ${PERCENTAGE_Y}%;
+      right: calc(${PERCENTAGE_Y}% + ${ARROW_SIZE + BOX_MARGIN}px);
```
[the remark this answers is not in the document]
Comment (pedrohmorais) [reply components/Tooltip/options.js]: It was overwritten by a new variable called `tipPositionDistance`

File: components/Tooltip/options.js [hunk 5]
```diff
@@ -46,22 +47,22 @@ const placementConfig = {
   tipPosition: {
     top: `
       left: ${PERCENTAGE_X}%;
-      bottom: ${PERCENTAGE_Y}%;
+      bottom: calc(${PERCENTAGE_Y}% + ${ARROW_SIZE + BOX_MARGIN}px);
       transform: translateX(-${PERCENTAGE_X}%);
     `,
     right: `
-      left: ${PERCENTAGE_Y}%;
+      left: calc(${PERCENTAGE_Y}% + ${ARROW_SIZE + BOX_MARGIN}px);
```
[the remark this answers is not in the document]
Comment (pedrohmorais) [reply components/Tooltip/options.js]: It was overwritten by a new variable called `tipPositionDistance`

**Pull request merged**